chainer_pytorch_migration/parameter.py

```diff
@@ -19,6 +18,10 @@ def __init__(self, link):
         super(LinkAsTorchModel, self).__init__()
         for n, p in sorted(link.namedparams()):
             self.__setattr__(n, ChainerParameter(p))
+        self.link = link
+
+    def forward(self, *input):
+        return self.link.forward(*input)
```

Comment:
`self.link` has parameters of `chainer.Parameter` type, so `self.link.forward` does not use desired parameters.

Reply:
There are two ways to avoid this issue:

1. Call `type(self.link).forward(self, *args, **kwargs)`.
2. Modify the attributes of `self.link` in place.

IMO, the latter one feels better because I want to prohibit bot use of torched model and the original Chainer model.

Reply:
`self.link.forward` should build the computation graph in chainer.

Reply:
You mean `LinkAsTorchModel` won't accept `torch.Tensor` as inputs?

Reply:
Exactly, the name is pretty confusing though

Reply:
I see. How about adding a comment and raising a `ValueError` when such inputs are given?

Reply:
adding a comment -> documenting it

Reply:
Thats a great idea, i will do it later today

Reply:
>Exactly, the name is pretty confusing though

Is that true? I think the inputs should usually be tensors, as `DataLoader` generates `Tensor`s.

Reply:
in the above example the dataloader collate_fn is set to return numpy arrays instead of torch tensors. in the migration guide its also written that when using this class, the computation graph should be done in chainer.

We can also convert the batch from tensor to array in this function but seems to magical for me.

I think we should change this class name to something less confusing but i cant come up with anything.